presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -442,14 +440,17 @@ public InputStreamSources createDictionaryStreamSources(Map<StreamId, Stream> st
                             .getColumnStatistics()
                             .getMinAverageValueSizeInBytes())
                     .sum();
-            rowGroupBuilder.add(createRowGroup(rowGroupId, rowOffset, rowsInGroup, minAverageRowBytes, valueStreams, checkpoints));
+            rowGroupBuilder.add(createRowGroup(rowGroupId, rowsInStripe, rowsInRowGroup, minAverageRowBytes, valueStreams, checkpoints));
         }
 
         return rowGroupBuilder.build();
     }
 
-    public static RowGroup createRowGroup(int groupId, int rowOffset, int rowCount, long minAverageRowBytes, Map<StreamId, ValueInputStream<?>> valueStreams, Map<StreamId, StreamCheckpoint> checkpoints)
+    @VisibleForTesting
+    static RowGroup createRowGroup(int groupId, long rowsInStripe, long rowsInRowGroup, long minAverageRowBytes, Map<StreamId, ValueInputStream<?>> valueStreams, Map<StreamId, StreamCheckpoint> checkpoints)
     {
+        long rowOffset = groupId * rowsInRowGroup;
```

Comment:
Maybe we should consider using `multiplyExact` to prevent overflow?

Reply:
Though this is unlikely, this is not in a performance sensitive path. I will change it to multiplyExact.